psi4/src/psi4/libscf_solver/uhf.cc

```diff
@@ -1069,4 +1069,32 @@ void UHF::compute_nos()
     }
 }
 
+
+std::shared_ptr<UHF> UHF::c1_deep_copy(std::shared_ptr<BasisSet> basis)
+{
+    std::shared_ptr<Wavefunction> wfn = Wavefunction::c1_deep_copy(basis);
+    auto hf_wfn = std::make_shared<UHF>(wfn, functional_, wfn->options(), wfn->psio());
+
+    // now just have to copy the matrices that UHF initializes
+    // include only those that are not temporary (some deleted in finalize())
+    if (Ca_) hf_wfn->Ca_ = Ca_subset("AO", "ALL");
+    if (Cb_) hf_wfn->Cb_ = Cb_subset("AO", "ALL");
+    if (Da_) hf_wfn->Da_ = Da_subset("AO");
+    if (Db_) hf_wfn->Db_ = Db_subset("AO");
+    if (Fa_) hf_wfn->Fa_ = Fa_subset("AO");
+    if (Fb_) hf_wfn->Fb_ = Fb_subset("AO");
+    if (epsilon_a_) hf_wfn->epsilon_a_ =
+        epsilon_subset_helper(epsilon_a_, nsopi_, "AO", "ALL");
+    if (epsilon_b_) hf_wfn->epsilon_b_ =
+        epsilon_subset_helper(epsilon_b_, nsopi_, "AO", "ALL");
+    // H_ ans X_ reset in the HF constructor, copy them over here
+    SharedMatrix SO2AO = aotoso()->transpose();
+    if (H_) hf_wfn->H_->remove_symmetry(H_, SO2AO);
+    if (X_) hf_wfn->X_->remove_symmetry(X_, SO2AO);
```

Comment:
If X is non symmetric, will this work here? Needs to be handled very carefully when we snip out small S eigenvalues.

Reply:
I don't see anything in the code that would fail if X is not symmetric... maybe there is something subtle I am missing though?  Perhaps someone else can check.

Reply:
Was DGAS warning about when the space is truncated for linear dependency?